### PR TITLE
UCT/RDMACM: decrease log level from error to diag

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
decrease log level from error to diag if IB device is not present

## Why ?
fixes https://github.com/open-mpi/ompi/issues/8367

### CODE REVIEW

**Review by yosefe — APPROVED** [review]
pls backport to v1.10.x

**Comment by yosefe** [comment]
@evgeny-leksikov can we add some test for no IB devices case?

**Comment by evgeny-leksikov** [comment]
> @evgeny-leksikov can we add some test for no IB devices case?

it's easy reproducible with `ucx_info -d` just need to have such machine in CI

**Comment by alinask** [comment]
won't this error still be printed?
https://github.com/openucx/ucx/blob/master/src/ucp/core/ucp_worker.c#L1435

**Comment by evgeny-leksikov** [comment]
> won't this error still be printed?
> https://github.com/openucx/ucx/blob/master/src/ucp/core/ucp_worker.c#L1435

nice catch, missed second print from UCP

**Comment by yosefe** [comment]
> > won't this error still be printed?
> > https://github.com/openucx/ucx/blob/master/src/ucp/core/ucp_worker.c#L1435
> 
> nice catch, missed second print from UCP

pls add some test for it, for example running gtest after deleting all components in libdir/ucx including libuct_ib.so and expecting no errors

**Comment by evgeny-leksikov** [comment]
> pls add some test for it, for example running gtest after deleting all components in libdir/ucx including libuct_ib.so and expecting no errors

it does not work:
`./build-devel/test/gtest/gtest: error while loading shared libraries: libuct_ib.so.0: cannot open shared object file: No such file or directory`
I will see how we can test it other way

**Comment by yosefe** [comment]
> it does not work:
> `./build-devel/test/gtest/gtest: error while loading shared libraries: libuct_ib.so.0: cannot open shared object file: No such file or directory`
> I will see how we can test it other way

maybe run small test on docker which does not have IB

**Comment by evgeny-leksikov** [comment]
@petro-rudenko could you please to add the following test to CI after merging this PR:
```
$ ucx_info -e -u t
#
# UCP endpoint
#
#               peer: hostname:148098
#                 lane[0]:  2:self/memory.0 md[2]           -> md[2]/self     am am_bw#0
#                 lane[1]:  6:knem/memory.0 md[5]           -> md[5]/knem     rma_bw#0
#
#                tag_send: 0..<egr/short>..8185..<egr/bcopy>..8192..<rndv>..(inf)
#            tag_send_nbr: 0..<egr/short>..8185..<egr/bcopy>..262144..<rndv>..(inf)
#           tag_send_sync: 0..<egr/short>..8185..<egr/bcopy>..8192..<rndv>..(inf)
#
#                  rma_bw: mds [5] rndv_rkey_size 42
#
$ echo $?
0
```
need to build and run UCX in container with all IB dependencies but without any IB devices
before this change it looks like:
```
$ ucx_info -e -u t
[1611759261.939723] [hostname:154490:0]      ucp_worker.c:1437 UCX  ERROR failed to open CM on component rdmacm with status No such device
<Failed to create UCP worker>
$ echo $?
242
```
pls note the return status of `ucx_info` is added in this PR as well to simplify testing, but need to be sure if this change will be merged after code review :)

**Comment by petro-rudenko** [comment]
@evgeny-leksikov 
https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=12975&view=logs&j=9ab45df2-ec5c-5734-36d7-99afc4959d97&t=68a66ff2-ea32-5af6-5ae4-1e6f59fc9498

It returns 0 even now

**Comment by petro-rudenko** [comment]
Should it be configured with debug ?

**Comment by evgeny-leksikov** [comment]
> @evgeny-leksikov
> https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=12975&view=logs&j=9ab45df2-ec5c-5734-36d7-99afc4959d97&t=68a66ff2-ea32-5af6-5ae4-1e6f59fc9498
> 
> It returns 0 even now

yes, error code added in this PR

**Comment by evgeny-leksikov** [comment]
> Should it be configured with debug ?

should work for release as well

**Comment by petro-rudenko** [comment]
Here's what it now prints in container:
```
```
/usr/bin/bash --noprofile --norc /__w/_temp/d1a5824f-e45a-4fb6-b62a-bcc6665bd31c.sh
#
# UCP endpoint
#
#               peer: <no debug data>
#                 lane[0]:  2:self/memory.0 md[2]           -> md[2]/self     am am_bw#0
#                 lane[1]:  4:tcp/eth0.0 md[3]              -> md[3]/tcp      rma_bw#0
#                 lane[2]:  3:tcp/lo.0 md[3]                -> md[3]/tcp      rma_bw#1 wireup
#
#                tag_send: 0..<egr/short>..8185..<egr/bcopy>..8192..<rndv>..(inf)
#            tag_send_nbr: 0..<egr/short>..8185..<egr/bcopy>..262144..<rndv>..(inf)
#           tag_send_sync: 0..<egr/short>..8185..<egr/bcopy>..8192..<rndv>..(inf)
#
#                  rma_bw: mds [3] rndv_rkey_size 10

```

**Comment by evgeny-leksikov** [comment]
> > @evgeny-leksikov
> > https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=12975&view=logs&j=9ab45df2-ec5c-5734-36d7-99afc4959d97&t=68a66ff2-ea32-5af6-5ae4-1e6f59fc9498
> > It returns 0 even now
> 
> yes, error code added in this PR

there is no rdmacm:
```
UCT modules:   < ib cma >

```
should be
```
UCT modules:   < ib rdmacm cma knem >
```

File: src/tools/info/proto_info.c [hunk 1]
```diff
@@ -93,10 +93,10 @@ static void print_resource_usage(const resource_usage_t *usage_before,
     printf("#\n");
 }
 
-void print_ucp_info(int print_opts, ucs_config_print_flags_t print_flags,
-                    uint64_t ctx_features, const ucp_ep_params_t *base_ep_params,
-                    size_t estimated_num_eps, size_t estimated_num_ppn,
-                    unsigned dev_type_bitmap, const char *mem_size)
+int print_ucp_info(int print_opts, ucs_config_print_flags_t print_flags,
```
Comment (yosefe) [inline src/tools/info/proto_info.c:96]: ucs_status_t

File: src/tools/info/proto_info.c [hunk 2]
```diff
@@ -113,7 +115,7 @@ void print_ucp_info(int print_opts, ucs_config_print_flags_t print_flags,
 
     status = ucp_config_read(NULL, NULL, &config);
     if (status != UCS_OK) {
-        return;
+        return status;
```
Comment (yosefe) [inline src/tools/info/proto_info.c:118]: goto out

**Review by yosefe — COMMENTED** [review]
pls squash after fix

**Review by yosefe — APPROVED** [review]

**Pull request merged**